src/go/metadata/metadata_fetcher_test.go

```diff
@@ -316,19 +315,3 @@ func TestFetchGCPAttributes(t *testing.T) {
 	}
 
 }
-
-func TestMetadataFetcherTimeout(t *testing.T) {
```

Comment:
Removed this because it's hard to test, we're using a hardcoded 30s. No option to change it to a lower value. Don't want unit test running for 30s.

Reply:
Maybe add a variable ImdsDeadLine to be used by metadata fetching, and test code change change it.

Reply:
I was thinking about if we should add a flag for users to adjust it.  But I don't see the drawback of setting a big timeout.  So,  default to a 30s timeout should be ok,  not need to add a new flag to adjust it.